app/lib/features/tasks/sheets/create_update_task_item.dart

```diff
@@ -0,0 +1,228 @@
+  final TextEditingController _taskNameController = TextEditingController();
+  final TextEditingController _taskDescriptionController =
+      TextEditingController();
+  final TextEditingController _taskDueDateController = TextEditingController();
+  DateTime? selectedDate;
+
+  @override
+  void initState() {
+    super.initState();
+    _taskNameController.text = widget.taskName;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _buildBody(context);
+  }
+
+  Widget _buildBody(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 20.0),
+      child: SingleChildScrollView(
+        child: Form(
+          key: _formKey,
+          child: Column(
+            mainAxisSize: MainAxisSize.min,
+            crossAxisAlignment: CrossAxisAlignment.stretch,
+            children: [
+              const SizedBox(height: 20),
+              const Divider(
+                indent: 150,
+                endIndent: 150,
+                thickness: 2,
+              ),
+              const SizedBox(height: 20),
+              Text(
+                L10n.of(context).addTask,
+                textAlign: TextAlign.center,
+                style: Theme.of(context).textTheme.titleMedium,
+              ),
+              const SizedBox(height: 20),
+              _widgetTaskName(),
+              const SizedBox(height: 20),
+              _widgetDescriptionName(),
+              const SizedBox(height: 20),
+              _widgetDueDate(),
+              const SizedBox(height: 20),
+              _widgetAddButton(),
+              const SizedBox(height: 20),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _widgetTaskName() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          L10n.of(context).taskName,
+          style: Theme.of(context).textTheme.bodySmall,
+        ),
+        const SizedBox(height: 5),
+        TextFormField(
+          autofocus: true,
+          decoration: InputDecoration(
+            hintText: L10n.of(context).name,
+          ),
+          autovalidateMode: AutovalidateMode.onUserInteraction,
+          controller: _taskNameController,
+          validator: (value) => (value?.isNotEmpty == true)
+              ? null
+              : L10n.of(context).pleaseEnterAName,
+        ),
+      ],
+    );
+  }
+
+  Widget _widgetDescriptionName() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          L10n.of(context).description,
+          style: Theme.of(context).textTheme.bodySmall,
+        ),
+        const SizedBox(height: 5),
+        TextFormField(
+          decoration: InputDecoration(
+            hintText: L10n.of(context).description,
+          ),
+          minLines: 4,
+          maxLines: 4,
+          controller: _taskDescriptionController,
+        ),
```

Comment:
isn't this a fully markup supporting field? We should use the appflowy editor here...

Reply:
Same as above.